src/python_testing/matter_testing_support.py

```diff
@@ -877,6 +878,45 @@ def get_test_desc(self, test: str) -> str:
         except AttributeError:
             return test
 
+    # Sends an out-of-band command to a Matter app
+    def write_to_app_pipe(self, command):
+        """
+        Use the following environment variables:
+
+         - LINUX_DUT_IP 
+            * if not provided, the Matter app is assumed to run on the same machine as this test,
+              such as during CI, and the commands are sent to it using a local named pipe
+            * if provided, the commands for writing to the named pipe are forwarded to the DUT
+        - LINUX_DUT_UNAME
+            * if LINUX_DUT_IP is provided, use this for the DUT user name
+            * if LINUX_DUT_NAME is not provided, a default value is used ("root")
+            * If a remote password is needed, set up ssh keys to ensure that this script can log in to the DUT without a password:
+                 + Step 1: If you do not have a key, create one using ssh-keygen
+                 + Step 2: Authorize this key on the remote host: run ssh-copy-id user@ip once, using your password
+                 + Step 3: From now on ssh user@ip will no longer ask for your password
+        """
+        import os
+        dut_ip = os.getenv('LINUX_DUT_IP')
+
+        if dut_ip is None:
+            with open(self.app_pipe, "w") as app_pipe:
+                app_pipe.write(command + "\n")
+            # TODO(#31239): remove the need for sleep
+            sleep(0.001)
+        else:
+            print(f"Using DUT IP address: {dut_ip}")
+
+            dut_uname = os.getenv('LINUX_DUT_UNAME')
+            if dut_uname is None:
+                dut_uname = "root"
+                print("Using default DUT user name (root)")
+            else:
+                print(f"Using DUT user name: {dut_uname}")
+
+            command_fixed = command.replace('\"', '\\"')
+            cmd = "echo \"%s\" | ssh %s@%s \'cat > %s\'" % (command_fixed, dut_uname, dut_ip, self.app_pipe)
```

Comment:
this assumes we can send stuff passwordless right?

Please make sure there is some documentation under docs describing these settings and how it works

Reply:
There is documentation about this and how to set up an ssh key on line 893. Is that enough?

Reply:
Could we add it somewhere in https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md as well?

Reply:
I have added documentation on how to use this in [this commit](https://github.com/project-chip/connectedhomeip/pull/34950/commits/0c2650a6d42923dffaba59d20bfa0550a5b4dfbf).

Reply:
See as @andy31415 is on holiday until the 30th, I'm going to resolve this comment. If the documentation is not adequate we'll raise another ticket.